### PR TITLE
Add new dispatcher service that handles background errors

### PR DESCRIPTION
**Pull request by hannesg**

### CODE REVIEW

File: services/rollout-service/pkg/service/dispatcher.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	return nil
+}
+
+func (r *Dispatcher) sendEvent(ctx context.Context, k Key, version *versions.VersionInfo, ev *v1alpha1.ApplicationWatchEvent) {
+	r.sink.ProcessArgoEvent(ctx, ArgoEvent{
+		Application:      k.Application,
+		Environment:      k.Environment,
+		SyncStatusCode:   ev.Application.Status.Sync.Status,
+		HealthStatusCode: ev.Application.Status.Health.Status,
+
```
Comment (sven-urbanski-freiheit-com) [inline services/rollout-service/pkg/service/dispatcher.go:117]: nitpick: unnecessary newline prevents indentation

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/rollout-service/pkg/service/dispatcher.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			revision: revision,
+			version:  version,
+		}
+		delete(r.unknown, k)
+		return version
+	}
+	// 3. Put this in the unknown queue and trigger the channel
+	r.unknown[k] = ev
+	select {
+	case r.unknownCh <- struct{}{}:
```
[the remark this answers is not in the document]
Comment (hannesg) [reply services/rollout-service/pkg/service/dispatcher.go:105]: I also had this idea but I didn't have a good idea for the channel capacity. Also we are only interested in the last event for every application+environment. If we use a channel, then we have to find a way to discard old entries.

[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply services/rollout-service/pkg/service/dispatcher.go:105]: ok then let's keep it as it is

File: services/rollout-service/pkg/service/dispatcher.go (hunk 1 above)
Comment (hannesg) [reply services/rollout-service/pkg/service/dispatcher.go]: Good idea. Done!

**Pull request merged**